manage.py

```diff
@@ -46,10 +53,24 @@ def test(all=False):
 def scrape():
     """Start an infinte loop to scrape & categorize articles."""
 
+    q = Queue('default', connection=conn)
     while True:
         logging.info("Starting scrape...")
-        scrape_and_categorize_articles()
-        logging.info("Finished scrape. Let's run it back...")
+        job = q.enqueue(scrape_and_categorize_articles)
```

Comment:
I don’t think it matters much. The limiting factor is really categorization and once that starts, it more than trumps the time to scrape anyway. Scraping 10 new feeds from scratch only took 3.5mins, but categorization (which I stopped after 4 mins) would take much longer that you’d have to wait for it anyway.

From: Alex Gerstein <notifications@github.com>
Reply: sourcemash/Sourcemash <reply@reply.github.com>>
Date: April 14, 2015 at 10:12:19 PM
To: sourcemash/Sourcemash <sourcemash@noreply.github.com>>
Cc: Scott Gladstone <sgladstone11@gmail.com>>
Subject:  Re: [Sourcemash] Scrape one feed: redis queue (#195)  

In manage.py:

>      while True:
>          logging.info("Starting scrape...")
> -        scrape_and_categorize_articles()
> -        logging.info("Finished scrape. Let's run it back...")
> +        job = q.enqueue(scrape_and_categorize_articles)
Instead of making the full scrape one task, it might help to break it down into feed scrapes and item categorizations. This way, when we add an urgent scrape to the front of the queue, it can be completed sooner.

—
Reply to this email directly or view it on GitHub.